packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,143 @@
+import '@aws-cdk/assert/jest';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as core from '@aws-cdk/core';
+import * as path from 'path';
+import * as inc from '../lib';
+import * as futils from '../lib/file-utils';
+
+// tslint:disable:object-literal-key-quotes
+/* eslint-disable quotes */
+
+describe('CDK Include', () => {
+  let stack: core.Stack;
+
+  beforeEach(() => {
+    stack = new core.Stack();
+  });
+
+  test("can ingest a template with nested stacks", () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: _testTemplateFilePath('parent-export-stack.json'),
+      nestedStacks: {
+        "ChildStack": {
+          templateFile: _testTemplateFilePath('child-import-stack.json'),
+          nestedStacks: {
+            "GrandChildStack": {
+              templateFile: _testTemplateFilePath('grandchild-import-stack.json'),
+            },
+          },
+        },
+        "AnotherChildStack": {
+          templateFile: _testTemplateFilePath('child-import-stack.json'),
+          nestedStacks: {
+            "GrandChildStack": {
+              templateFile: _testTemplateFilePath('grandchild-import-stack.json'),
+            },
+          },
+        },
+      },
+    });
+
+    expect(stack).toMatchTemplate(
+      loadTestFileToJsObject(('parent-stack.expected.json')),
+    );
+
+    expect(parentTemplate.getNestedStack('ChildStack')).toMatchTemplate(
+      loadTestFileToJsObject(('child-stack.expected.json')),
+    );
+
+    expect(parentTemplate.getNestedStackTemplate('ChildStack').getNestedStack('GrandChildStack')).toMatchTemplate(
+      loadTestFileToJsObject(('grandchild-stack.expected.json')),
+    );
+
+    expect(parentTemplate.getNestedStack('AnotherChildStack')).toMatchTemplate(
+      loadTestFileToJsObject(('another-child-stack.expected.json')),
+    );
+
+    expect(parentTemplate.getNestedStackTemplate('AnotherChildStack').getNestedStack('GrandChildStack')).toMatchTemplate(
+      loadTestFileToJsObject(('grandchild-stack.expected.json')),
+    );
+  });
```

Comment:
Gotta say, I don't love this test. We already have a mechanism for doing snapshot tests. Can you use that instead?

Take a look at the existing nested stacks support tests to see how you can have assertions with multiple stacks involved. Example: https://github.com/aws/aws-cdk/blob/31af334ae240c0e44df39f397f3c81ed6c6cd9af/packages/%40aws-cdk/aws-cloudformation/test/integ.nested-stacks-nested-export-to-sibling.expected.json#L1-L106 .